spring-kafka-test/src/main/java/org/springframework/kafka/test/EmbeddedKafkaBroker.java

```diff
@@ -546,11 +544,12 @@ public void consumeFromAnEmbeddedTopic(Consumer<?, ?> consumer, String topic) {
 	 * @param topicsToConsume the topics.
 	 */
 	public void consumeFromEmbeddedTopics(Consumer<?, ?> consumer, String... topicsToConsume) {
-		HashSet<String> diff = new HashSet<>(Arrays.asList(topicsToConsume));
-		diff.removeAll(new HashSet<>(this.topics));
-		assertThat(this.topics)
-				.as("topic(s):'" + diff + "' are not in embedded topic list")
-				.containsAll(new HashSet<>(Arrays.asList(topicsToConsume)));
+		List<String> notEmbedded = Arrays.stream(topicsToConsume)
+				.filter(topic -> !this.topics.contains(topic))
+				.collect(Collectors.toList());
+		if (notEmbedded.size() > 0) {
+			throw new IllegalStateException("topic(s):'" + notEmbedded + "' are not in embedded topic list");
```

Comment:
Shouldn't we JavaDoc this behavior? It's not a big breaking change though, but would be great to have some note about this exception.

Thanks